crates/net/dns/src/lib.rs

```diff
@@ -412,12 +412,60 @@ fn convert_enr_node_record(enr: &Enr<SecretKey>) -> Option<DnsNodeRecordUpdate>
 mod tests {
     use super::*;
     use crate::tree::TreeRootEntry;
-    use alloy_rlp::Encodable;
+    use alloy_rlp::{Decodable, Encodable};
     use enr::EnrKey;
     use reth_primitives::{Chain, Hardfork, MAINNET};
     use secp256k1::rand::thread_rng;
     use std::{future::poll_fn, net::Ipv4Addr};
 
+    #[test]
+    fn test_convert_enr_node_record() {
+        // rig
+        let secret_key = SecretKey::new(&mut secp256k1::rand::thread_rng());
+        let enr = Enr::builder()
+            .ip("127.0.0.1".parse().unwrap())
+            .udp4(9000)
+            .tcp4(30303)
+            .add_value(b"eth", &MAINNET.latest_fork_id())
+            .build(&secret_key)
+            .unwrap();
+
+        // test
+        let node_record_update = convert_enr_node_record(&enr).unwrap();
+
+        assert_eq!(node_record_update.node_record.address, "127.0.0.1".parse::<IpAddr>().unwrap());
+        assert_eq!(node_record_update.node_record.tcp_port, 30303);
+        assert_eq!(node_record_update.node_record.udp_port, 9000);
+        assert_eq!(node_record_update.fork_id, Some(MAINNET.latest_fork_id()));
+        assert_eq!(node_record_update.enr, enr);
+    }
+
+    #[test]
+    fn test_decode_and_convert_enr_node_record() {
+        // rig
+        let secret_key = SecretKey::new(&mut secp256k1::rand::thread_rng());
+        let enr = Enr::builder()
+            .ip("127.0.0.1".parse().unwrap())
+            .udp4(9000)
+            .tcp4(30303)
+            .add_value(b"eth", &MAINNET.latest_fork_id())
```

Comment:
could you try this with a different field, like b"dummy"

Reply:
do you mind experimenting locally? ☺️

Reply:
```
.add_value(b"dummy", &MAINNET.latest_fork_id())
.add_value(b"eth", &MAINNET.latest_fork_id())
``` 

this fails decoding because this is buggy:

https://github.com/sigp/enr/blob/df661bb47fff0e76d958b2cf73efa1e4f60918de/src/lib.rs#L1048-L1051

this should be to preserve the encoded value

```rust
                _ => {
                    let other_header = Header::decode(payload)?;
                    let value = &payload[..other_header.payload_length];
                    let mut out = Vec::with_capacity(other_header.length());
                    payload.advance(other_header.payload_length);
                    other_header.encode(&mut out);
                    out.extend_from_slice(value);
                    out
                }
```

but even with this fix, decoding forkid only works for DNS records with:

```rust
    let maybe_fork_id = enr.get(b"eth")?;
    let decoded_fork_id = ForkId::decode(&mut &maybe_fork_id[..]).ok();
```

no idea why...